### DISCUSSION TITLE
Multi-GPUs and CPU in DUALSPHYSICS_V5.4_BETA

### DISCUSSION THREAD

**Discussion by jialing95** [DualSPHysics/DualSPHysics#222] [DUALSPHYSICS_V5.4_BETA]
Hi, folks

I'm faced with tough problems when doing large-scale simulations for my PhD proposal.

I was wondering if DualSPHysics supports multi-GPUs in v5.4 now? If so, what's the maximum number of GPUs?
[(https://github.com/DualSPHysics/DualSPHysics/discussions/217#discussioncomment-10236169 
)]

I also want to know how many CPUs I could use in v5.4. Any details about the threads, cores, etc?

Thank you so much!!

**Comment by Claxomere** [comment]
Hi,
No multi-GPU yet ! No idea for CPU core.
Good luck

**Comment by alexbexe** [comment]
We are planing to release a new version 5.4 next March 2025.
However, regarding multiGPU, that will be available as one of the main novelties in the release of v6.0, expected for 2026.

Regarding CPUs? OpenMP allows you to use all different threads or cores available in your CPU device.

Regards

**Reply by ZXCCA** [reply]
Hi,

Thank you for all the works have done in this project.
In the comment you replied before mentioning about the feature of multi-GPU in the future version. However, is it possible for user to recompile the source code by adding this feature? Or do you think it is too complicated to achieve?

Best regards

**Comment by jmdalonso** [comment]
Implementing a multi-GPU solution that supports the main functionalities of DualSPHysics and is efficient is not a simple task. Changes to the pre- and post-processing tools that accompany the multi-GPU code are also necessary.
A beta version of DualSPHysics multi-GPU is scheduled for distribution to the attendees of the 8th DualSPHysics Workshop (https://dual.sphysics.org/8thworkshop/) in January 2026.

Best regards
